src/training_hub/algorithms/embedding_sft.py

```
@@ -0,0 +1,334 @@
+
+def _to_pair_dataset(dataset):
+    """Convert a label-based dataset to (anchor, positive) pairs for MNRL."""
+    from datasets import Dataset
+    from itertools import combinations
+
+    groups = {}
+    for row in dataset:
+        groups.setdefault(row["label"], []).append(row["text"])
+
+    pairs = []
+    for label, texts in groups.items():
```

Ruff (0.16.0)

[warning] Loop control variable `label` not used within loop body

Rename unused `label` to `_label`

(B007)